ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2171,22 +2172,32 @@ private Materialization getMaterializationInvalidationInfo(MaterializedViewMetad
         materialization.setSourceTablesCompacted(true);
         return materialization;
       }
-      Boolean b = storageHandler.hasAppendsOnly(
-          table, mvSnapshot.getTableSnapshots().get(table.getFullyQualifiedName()));
-      if (b == null) {
+
+      boolean hasAppendsOnly = true;
```

Comment:
why do we need 1 more flag, can we leave with 1?

Reply:
Refactored to rely on only one flag

Reply:
@kasakrisz, have you submitted the change?

Reply:
sorry, I just pushed the missing commit